processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
Now I'm confused as well: I didn't intend to pull in the automatic inheritance for mapping methods in the same mapper again that we had before we introduced ```@InheritConfiguration```. My gut-feeling (no examples available) is that by also "auto"-inheriting mapping-configuration from the current mapper's methods, we increase the likelyhood of ambiguities and, and thus render the AUTO flag useless. 

Oh, I have an example after all: imagine you specify mappings for base types (abstract or not, these are types that you wouldn't use on their own) in the MapperConfig interface and want to auto-inherit them. In your mapper method, you have ```Car toCar(CarDto s)``` and ```void intoCar(CarDto s, @MappingTarget Car t)``` - if the current logic of auto-inheritance was applied, but would include the methods in the current mapper as well, then each of the methods would find two candidates for the auto-inheritance: the one regarding the base types in the MapperConfig interface and the sibling method defined in the mapper.

> Defining methods on config classes should be a rare cases.

I also see it more like Sjaak: I would have one or two prototype methods covering the base-types in the mapper-config interface to be consumed with auto-inheritance. All those other mapper classes use that mapperConfig and could then only mind their own business. Our Dtos / Entities don't span out a class hierarchy where it would make sense to have mapping methods for types "in between".
So from my perspective, inheriting mappings from methods with super-types within the same mapper would be the rare case. Funny :smile: ...

Now what do we do here... From where I'm standing, this is a change that could use some more thoughts... E.g. an additional stragety to allow being configured? Or would it make sense to select the "nearest" match for the auto-inheritance and continue inheriting transively instead of complaining about ambiguities too early?

Would it be OK to discuss / change that within a separate issue?